### PR TITLE
Issue 61 : Frontend - Add details in the datasource card

### LINKED ISSUE #61: Add datasource details in the list
When listing data sources, it will be useful to see some details like the cluster identifier, the database or the user as a description:

unnamed

Similar to postgresql:

unnamed (2)

### PR DESCRIPTION
**Pull request by yaelleC**
Fixes #61 

## Description
When listing datasources, a card is composed of a Heading ('Amazon Redshift'), a Figure (the logo) and some `meta` info. Meta include the datasource `typeName`, `url` and `isDefault`.

This PR uses the endpoint created by #129 to use a dropdown instead of an input field and populate the `url` field.

It also upgrades `aws-sdk` to use `allowCustomValue` in case the user does not have permission to fetch the clusters

On the `managed Secret` tab, the dropdown is hidden and a plain text field is shown. The reason for that is that since the dropdown gets its options populated dynamically, when selecting a secret it would not show the clusterID associated with it (as it's a disabled dropdown with no valid options)

### Screenshots

No change to the `managed secret` UI
Dropdown instead of text input for `Temp credential` UI
Details added to card in datasources list view

| Before | After |
|--------|-------|
|  Screenshot 2022-02-22 at 10 27 20    | Screenshot 2022-02-22 at 10 27 45   |
|  Screenshot 2022-03-07 at 20 03 24 | Screenshot 2022-03-07 at 20 03 36 |
| Screenshot 2022-03-17 at 12 47 28  | Screenshot 2022-03-17 at 12 45 41 |

| If the user does not have permission to fetch the clusters, she can create one | And the `AWS Secret Manager` view works as expected |
|--------|-------|
| Screenshot 2022-03-18 at 09 54 39 | Screenshot 2022-03-18 at 09 54 53 |

## Test coverage
Running: `yarn test:coverage:change`

```
-------------------|---------|----------|---------|---------|-------------------
File               | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-------------------|---------|----------|---------|---------|-------------------
  ConfigEditor.tsx |   91.23 |    79.31 |   94.74 |      90 | 39-44,95-96 
-------------------|---------|----------|---------|---------|-------------------
```

### CODE REVIEW

File: src/ConfigEditor/ConfigEditor.test.tsx
Comment (yaelleC) [inline src/ConfigEditor/ConfigEditor.test.tsx:7]: auto reorg

File: src/ConfigEditor/ConfigEditor.tsx
Comment (yaelleC) [inline src/ConfigEditor/ConfigEditor.tsx:5]: auto reorg

File: src/ConfigEditor/ConfigEditor.tsx
Comment (yaelleC) [inline src/ConfigEditor/ConfigEditor.tsx:12]: auto re-org

File: src/ConfigEditor/ConfigEditor.tsx
Comment (yaelleC) [inline src/ConfigEditor/ConfigEditor.tsx:75]: this returns a database, should we make the input disable and fill automatically based on clusterId selected?

Comment (andresmgot) [reply src/ConfigEditor/ConfigEditor.tsx:75]: I don't think we should, that's the default database, it's possible to create a different one in the same cluster and use that one instead. I would also not display this URL in this selector as the description for the same reason.

File: src/ConfigEditor/ConfigEditor.tsx
Comment (andresmgot) [inline src/ConfigEditor/ConfigEditor.tsx:156]: we should still show this field as disabled when `useManagedSecret`

**Review by andresmgot — COMMENTED** [review]
Thanks! Just a couple of minor comments here.

Also, would you able to create a PR for the SDK allowing to set other properties? ([context](https://github.com/grafana/redshift-datasource/pull/127#discussion_r818458039)) We can open a different issue for that if you don't have time now to do it atm.

**Comment by andresmgot** [comment]
BTW, you need to adapt the e2e test (smoke.test.ts) to this change. See how it's done for `e2eSelectors.ConfigEditor.ManagedSecret` as an example. If you need any help running the e2e tests locally (or you have any question) let me know!

File: src/ConfigEditor/ConfigEditor.tsx
Comment (yaelleC) [reply src/ConfigEditor/ConfigEditor.tsx:75]: Thanks!
So I should use the database entered in the db field to create the url in the datasource card instead of this default one?

Comment (andresmgot) [reply src/ConfigEditor/ConfigEditor.tsx:75]: yes, that would make sense

File: src/ConfigEditor/ConfigEditor.tsx
Comment (yaelleC) [reply src/ConfigEditor/ConfigEditor.tsx:156]: weird, when I have `disabled={useManagedSecret}` it does not disable the select. 

When I inspect, I see that although `ConfigSelect` has `disabled=true` it's child `ResourceSelector` has `disabled=false`. 

It seems to be because we override disable there: https://github.com/grafana/grafana-aws-sdk-react/blob/6a051089daf5c577297f4bf695909b4da0194f9f/src/sql/ConfigEditor/ConfigSelect.tsx#L47 

I will fix that as part of the SDK PR 🙂

File: cypress/integration/smoke.spec.ts
Comment (andresmgot) [inline cypress/integration/smoke.spec.ts:58]: you need to click first, wait for the clusters to load and then type here (see lines 113:115 for the example with secrets)

**Review by andresmgot — APPROVED** [review]
LGTM! but it would be better if we fix the bugs in the SDK before merging this PR (so `main` is still releaseable, just in case)

**Comment by yaelleC** [comment]
> LGTM! but it would be better if we fix the bugs in the SDK before merging this PR (so main is still releaseable, just in case)

Agreed!

File: src/ConfigEditor/ConfigEditor.test.tsx
Comment (yaelleC) [inline src/ConfigEditor/ConfigEditor.test.tsx:56]: These tests would always pass as even `hidden` or `disabled` components will be in the document 😄 

I've made them stricter to check which inputs are disabled and which are visible

File: src/ConfigEditor/ConfigEditor.tsx
Comment (yaelleC) [inline src/ConfigEditor/ConfigEditor.tsx:106]: I tried to use memoize here to avoid fetching unnecessary clusters (if clusterID is still the same) but I could not get it to work so I added the memo check myself 😉

File: src/ConfigEditor/ConfigEditor.tsx
Comment (yaelleC) [inline src/ConfigEditor/ConfigEditor.tsx:164]: A dropdown for better use experience in `temporary Creds`

File: src/ConfigEditor/ConfigEditor.tsx
Comment (yaelleC) [inline src/ConfigEditor/ConfigEditor.tsx:175]: A text input to display clusterID for `managed Secret`

File: cypress/integration/smoke.spec.ts
Comment (yaelleC) [inline cypress/integration/smoke.spec.ts:115]: any idea why this line fails? 🤔 
I figured we don't need it as the next few ones types in the expected secret and selects it but leaving as commented out for visibility

File: src/ConfigEditor/ConfigEditor.tsx
Comment (yaelleC) [reply src/ConfigEditor/ConfigEditor.tsx:180]: Well spotted 😄

File: src/ConfigEditor/ConfigEditor.tsx
Comment (yaelleC) [reply src/ConfigEditor/ConfigEditor.tsx:137]: yes, I agree, I could not find a workaround but using state seems like a good idea, I'll try that

File: src/ConfigEditor/ConfigEditor.tsx
Comment (yaelleC) [reply src/ConfigEditor/ConfigEditor.tsx:73]: Added inside `fetchClusters` to return an empty array in catch

File: cypress/integration/smoke.spec.ts
Comment (yaelleC) [reply cypress/integration/smoke.spec.ts]: It's managedSecret, not clusters so permissions should not be an issue 🤷‍♀️ 
I can replicate locally, it is set up exactly the same way as the clusterID dropdown (which passes) - some puzzling behaviour! 🤔 

Screenshot 2022-03-17 at 22 28 32

File: src/ConfigEditor/ConfigEditor.tsx
Comment (yaelleC) [reply src/ConfigEditor/ConfigEditor.tsx:191]: Added issue: https://github.com/grafana/grafana-aws-sdk-react/issues/15

File: cypress/integration/smoke.spec.ts
Comment (andresmgot) [reply cypress/integration/smoke.spec.ts]: mmh, weird, anyway it seems stable now

**Review by andresmgot — APPROVED** [review]
:+1: :+1:

**Pull request merged**
